app/src/org/commcare/dalvik/activities/CommCareSetupActivity.java

```diff
@@ -244,7 +246,19 @@ private void uiStateScreenTransition() {
                 break;
 
             default:

```

Comment:
I don't like that you are re-instating the url chooser fragment in the default case, given that UiState.advanced is the dedicated UiState for being in the url chooser fragment. Couldn't you just set that when the chooser is launched and use it to properly restore?